> Code change is good, but please fix the commit message formatting.

I second the above and would also add the need to fix the failing checks in CI:

```console
FAILED: [code=1] modules/sof/CMakeFiles/modules_sof.dir/home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c.obj 
/usr/bin/clang -DCC_OPTIMIZE_FLAGS=\"\" -DKERNEL -DK_HEAP_MEM_POOL_SIZE=2048 -DXCC_TOOLS_VERSION=\"host\" -D_POSIX_THREAD_SAFE_FUNCTIONS=200809L -D__ZEPHYR__=1 -DRELATIVE_FILE=\"../sof/src/ipc/ipc3/host-page-table.c\" -I/home/runner/work/sof/sof/workspace/build-fuzz/zephyr/include/generated/zephyr -I/home/runner/work/sof/sof/workspace/zephyr/include -I/home/runner/work/sof/sof/workspace/build-fuzz/zephyr/include/generated -I/home/runner/work/sof/sof/workspace/zephyr/soc/native/inf_clock -I/home/runner/work/sof/sof/workspace/zephyr/boards/native/native_sim -I/home/runner/work/sof/sof/workspace/zephyr/scripts/native_simulator/common/src/include -I/home/runner/work/sof/sof/workspace/zephyr/scripts/native_simulator/native/src/include -I/home/runner/work/sof/sof/workspace/zephyr/subsys/portability/posix/c_lib_ext/getopt -I/home/runner/work/sof/sof/workspace/zephyr/drivers -I/home/runner/work/sof/sof/workspace/sof/zephyr/include -I/home/runner/work/sof/sof/workspace/zephyr/kernel/include -I/home/runner/work/sof/sof/workspace/zephyr/arch/host/include -I/home/runner/work/sof/sof/workspace/sof/src/include/sof/audio/module_adapter/iadk -I/home/runner/work/sof/sof/workspace/sof/src/include/sof/audio/module_adapter/library -I/home/runner/work/sof/sof/workspace/sof/src/audio/smart_amp/CONFIG_MAXIM_DSM -I/home/runner/work/sof/sof/workspace/sof/src/audio/smart_amp/include/dsm_api/inc -I/home/runner/work/sof/sof/workspace/sof/src/platform/posix/include -I/home/runner/work/sof/sof/workspace/sof/tools/rimage/src/include -I/home/runner/work/sof/sof/workspace/sof/src/include -I/home/runner/work/sof/sof/workspace/sof/src/arch/host/include -I/home/runner/work/sof/sof/workspace/sof/third_party/include -isystem /home/runner/work/sof/sof/workspace/zephyr/lib/libc/minimal/include -isystem /home/runner/work/sof/sof/workspace/zephyr/lib/libc/common/include -fno-strict-aliasing -O2 -imacros /home/runner/work/sof/sof/workspace/build-fuzz/zephyr/include/generated/zephyr/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fcolor-diagnostics --config=/home/runner/work/sof/sof/workspace/zephyr/cmake/toolchain/host/llvm/clang_libgcc.cfg -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-unused-but-set-variable -Wno-typedef-redefinition -Wno-deprecated-non-prototype -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wno-self-assign -Wno-initializer-overrides -Wno-section -Wno-gnu -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fstrict-overflow -Wno-vla -fmacro-prefix-map=/home/runner/work/sof/sof/workspace/sof/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/runner/work/sof/sof/workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/runner/work/sof/sof/workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -msse2 -mfpmath=sse -fvisibility=hidden -nostdinc -isystem /usr/lib/llvm-18/lib/clang/18/include -include /home/runner/work/sof/sof/workspace/zephyr/arch/posix/include/undef_system_defines.h -fno-builtin -fsanitize-recover=all -fPIC -fsanitize=address,fuzzer -std=c17 -fno-inline-functions -std=gnu99 -MD -MT modules/sof/CMakeFiles/modules_sof.dir/home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c.obj -MF modules/sof/CMakeFiles/modules_sof.dir/home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c.obj.d -o modules/sof/CMakeFiles/modules_sof.dir/home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c.obj -c /home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c
/home/runner/work/sof/sof/workspace/sof/src/ipc/ipc3/host-page-table.c:230:20: error: use of undeclared identifier 'PLATFORM_PAGE_TABLE_SIZE'
  230 |             ring->pages > PLATFORM_PAGE_TABLE_SIZE * 8 / 20) {
      |                           ^
1 error generated.
```

--- a/src/ipc/ipc3/helper.c
+++ b/src/ipc/ipc3/helper.c
@@ -302,7 +302,11 @@ static int comp_specific_builder(struct sof_ipc_comp *comp,
 		if (IPC_TAIL_IS_SIZE_INVALID(*proc))
 			return -EBADMSG;
 
-		if (proc->comp.hdr.size + proc->size > SOF_IPC_MSG_MAX_SIZE)
+		/* compare without adding the two host-supplied uint32_t values,
+		 * which could wrap and let an oversized proc->size pass
+		 */
+		if (proc->comp.hdr.size > SOF_IPC_MSG_MAX_SIZE ||
+		    proc->size > SOF_IPC_MSG_MAX_SIZE - proc->comp.hdr.size)
 			return -EBADMSG;
 
 		config->process.type = proc->type;

--- a/src/ipc/ipc3/host-page-table.c
+++ b/src/ipc/ipc3/host-page-table.c
@@ -17,6 +17,25 @@
 LOG_MODULE_DECLARE(ipc, CONFIG_SOF_LOG_LEVEL);
 
 /*
+ * Size in bytes of the DSP-side buffer that receives the compressed host page
+ * table. DSP targets define PLATFORM_PAGE_TABLE_SIZE; the host/library builds
+ * (e.g. testbench, fuzzer) allocate it as HOST_PAGE_SIZE in drivers/host/ipc.c
+ * and do not define PLATFORM_PAGE_TABLE_SIZE, so fall back to that.
+ */
+#ifndef PLATFORM_PAGE_TABLE_SIZE
+#define PLATFORM_PAGE_TABLE_SIZE HOST_PAGE_SIZE
+#endif
+
+/*
+ * The compressed page table stores one 20-bit entry per host page, so the
+ * number of pages whose entries fit in the page table buffer is
+ * (buffer bytes * 8 bits/byte) / 20 bits/page.
+ */
+#define HOST_PAGE_TABLE_BITS_PER_PAGE	20
+#define HOST_PAGE_TABLE_MAX_PAGES \
+	(PLATFORM_PAGE_TABLE_SIZE * 8 / HOST_PAGE_TABLE_BITS_PER_PAGE)
+
+/*
  * Parse the host page tables and create the audio DMA SG configuration
  * for host audio DMA buffer. This involves creating a dma_sg_elem for each
  * page table entry and adding each elem to a list in struct dma_sg_config.
@@ -215,6 +234,19 @@ int ipc_process_host_buffer(struct ipc *ipc,
 {
 	struct ipc_data_host_buffer *data_host_buffer;
 	int err;
+
+	/*
+	 * The host-supplied page count is used both to size DSP-side
+	 * allocations and to compute the DMA transfer length for the
+	 * compressed page table. Reject a count that would not fit in the
+	 * page table buffer before doing any arithmetic that could overflow
+	 * (ring->pages * 20). pages == 0 is also invalid and would underflow
+	 * the ring->size sanity check in ipc_parse_page_descriptors().
+	 */
+	if (ring->pages == 0 || ring->pages > HOST_PAGE_TABLE_MAX_PAGES) {
+		tr_err(&ipc_tr, "ipc: invalid page count %u", ring->pages);
+		return -EINVAL;
+	}
 
 	data_host_buffer = ipc_platform_get_host_buffer(ipc);
 	dma_sg_init(elem_array);

